src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,53 +1,107 @@
-import { useState } from 'react';
+import classNames from 'classnames';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```

Comment:
Combine it into one state
```suggestion
const [formValues, setFormValues] = useState({
  title: '',
  description: '', 
  imgUrl: '',
  imdbUrl: '',
  imdbId: '',
});
```

and then you could have only one handler 
```
const handleChange = ((value: string, name: string) => setFormValues(prevState => ({
  ...prevState,
  [name]: value,
})));
```